validmind/unit_metrics/__init__.py

```diff
@@ -0,0 +1,192 @@
+
+    Args:
+        params (dict): The parameters to be serialized.
+
+    Returns:
+        str: A SHA-256 hash of the JSON string representation of the params.
+    """
+    # Convert the params to a JSON string, ensuring consistent ordering
+    params_json = json.dumps(params, sort_keys=True)
+
+    # Create a SHA-256 hash of the JSON string
+    hash_object = hashlib.sha256(params_json.encode())
+    return hash_object.hexdigest()
+
+def _serialize_model(model):
+    """
+    Generate a SHA-256 hash for a scikit-learn model based on its type and parameters.
+
+    Args:
+        model VMModel: The model to be serialized.
+
+    Returns:
+        str: A SHA-256 hash of the model's description.
+    """
+
+    if model is None:
+        # You can return a predefined hash, or handle it in another appropriate way
+        model_info = "none_model_hash"
+    else:
+        model_info = get_model_info(model)
+
+    model_json = json.dumps(model_info, sort_keys=True)
+
+    # Create a SHA-256 hash of the JSON string
+    hash_object = hashlib.sha256(model_json.encode())
+    return hash_object.hexdigest()
+
+def _serialize_dataset(dataset):
+    """
+    Serialize the description of the dataset input to a unique hash.
+
+    This function generates descriptive statistics of the dataset, converts
+    the description to a JSON string, and then creates a SHA-256 hash of the string
+    to ensure a unique identifier for the dataset's description.
+
+    Args:
+        dataset (pd.DataFrame): The dataset whose description is to be serialized.
+
+    Returns:
+        str: A SHA-256 hash of the JSON string representation of the dataset's description.
+    """
+    if isinstance(dataset._df, pd.DataFrame):
+        # Get the description of the DataFrame
+        columns = [dataset.target_column] + dataset.feature_columns
+        description = dataset._df[columns].describe()
+        # Convert the description DataFrame to a JSON string
+        description_json = json.dumps(
+            description.to_dict(orient="records"), sort_keys=True
+        )
+        # Create a SHA-256 hash of the JSON string
+        hash_object = hashlib.sha256(description_json.encode())
+        return hash_object.hexdigest()
+    else:
+        # If it's not a DataFrame, we cannot get a description
+        raise TypeError("The dataset provided is not a pandas DataFrame.")
+
+def _get_metric_class(metric_id):
+    """Get the metric class by metric_id
+
+    This function will load the metric class by metric_id.
+
+    Args:
+        metric_id (str): The full metric id (e.g. 'validmind.vm_models.test.v2.model_validation.sklearn.F1')
+
+    Returns:
+        Metric: The metric class
+    """
+
+    metric_module = importlib.import_module(f"{metric_id}")
+
+    class_name = metric_id.split(".")[-1]
+
+    # Access the class within the F1 module
+    metric_class = getattr(metric_module, class_name)
+
+    return metric_class
+
+def get_metric_value(metric_id, inputs=None):
```

Comment:
It looks like this is doing almost the same as `run_metric`? In any case this method is not being used by any function so let's remove it for now. We can always add it back once we understand what we need to do with it.

Reply:
Done